### PR TITLE
Added support to track pod crash/restart during sleep

### PR DESCRIPTION
**Pull request by yashashreesuresh**
This commit enables the pod crash/restart to be tracked during the wait time between each iteration. This prevents the Cerberus from missing pod crash/restarts when the pod enters the running phase before the next iteration.

sleep_tracker1

Fixes: #51

### CODE REVIEW

**Comment by yashashreesuresh** [comment]
Please have a look @chaitanyaenr @mffiedler

File: cerberus/kubernetes/client.py [hunk 1]
```diff
@@ -80,6 +82,41 @@ def check_sdn_namespace():
...
+                for container in pod_status.init_container_statuses:
+                    pod_restart_count += container.restart_count
+            if pods_tracker[pod]:
+                if pods_tracker[pod]["creation_timestamp"] != pod_creation_timestamp or \
+                    pods_tracker[pod]["restart_count"] != pod_restart_count:
+                    crashed_restarted_pods.append(pod)
+                    pods_tracker[pod]["creation_timestamp"] = pod_creation_timestamp
+                    pods_tracker[pod]["restart_count"] = pod_restart_count
+            else:
+                crashed_restarted_pods.append(pod)
```
Comment (mffiedler) [inline cerberus/kubernetes/client.py:114]: The list of crashed/restarted pods is currently the pod name.  A possible improvement is to append <pod_namspace>:<pod_name> to the list to make it clear where the pod is.   This is minor though, I am fine if we move forward like it is now.

File: start_cerberus.py [hunk 2]
```diff
@@ -165,6 +167,10 @@ def main(cfg):
                                            watch_namespaces_status, failed_nodes,
                                            failed_pods_components)
 
+            if iteration != 1 and crashed_restarted_pods:
+                logging.info("Pods that were crashed/restarted during the sleep: %s"
```
Comment (mffiedler) [inline start_cerberus.py:171]: Suggestion:  "Pods that were crashed/restarted during iteration %s : %s" % (iteration, crashed_restarted_pods)"

**Comment by mffiedler** [comment]
Not sure why the duplicate comment.  In any case my comments are nitpicks.  I tested this and it looks good to me.

File: cerberus/kubernetes/client.py [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
 from kubernetes import client, config
 from kubernetes.client.rest import ApiException
 import cerberus.invoke.command as runcommand
-import yaml
+
 
 # Load kubeconfig and initialize kubernetes python client
 def initialize_clients(kubeconfig_path):
-    global cli
+    global cli, pods_tracker
```
Comment (chaitanyaenr) [inline cerberus/kubernetes/client.py:11]: Maybe initialize pods_tracker outside this function as it's meant for initializing client?

File: cerberus/kubernetes/client.py [hunk 4]
```diff
@@ -80,6 +82,41 @@ def check_sdn_namespace():
         please specify the correct networking namespace in config file")
 
 
+def namespace_sleep_track(namespace):
+    global pods_tracker
+    pods = list_pods(namespace)
+    crashed_restarted_pods = []
+    for pod in pods:
+        try:
+            pod_info = cli.read_namespaced_pod_status(pod, namespace,
```
Comment (chaitanyaenr) [inline cerberus/kubernetes/client.py:91]: Make this a separate function and call it to provide the given pod status? This way we can reuse it in other places as well. Thoughts?

File: cerberus/kubernetes/client.py [hunk 5]
```diff
@@ -80,6 +82,41 @@ def check_sdn_namespace():
         please specify the correct networking namespace in config file")
 
 
+def namespace_sleep_track(namespace):
```
Comment (chaitanyaenr) [inline cerberus/kubernetes/client.py:85]: Let's add a comment to describe the functionality of this function.

**Comment by yashashreesuresh** [comment]
I have made all the above changes. PTAL @mffiedler @chaitanyaenr 
sleep_tracker

**Comment by chaitanyaenr** [comment]
Nice job @yashashreesuresh.

**Pull request merged**